system/src/Grav/Framework/Form/Interfaces/FormInterface.php

```diff
@@ -131,14 +131,14 @@ public function getFlash();
 
     /**
      * @param ServerRequestInterface $request
-     * @return $this
```

Comment:
$this was actually the correct return value, but current PHP versions do not support covariant return values.

Reply:
Same applies to all the later ones, too. The covariance will be supported in PHP 7.3 and we should add those type hints where ever we will need to change those in the future.

Reply:
Covariant returns are not until PHP 7.4 so at this point you can't return a more specific type from an interface method like this.

Reply:
I meant PHP 7.4... I must have misspelt there or perhaps I had a brain-fart.

Nevertheless, I can for sure return more specific type as it is very useful to use covariance there even if PHP does not yet support it as a return type. This allows me to access all the methods from my more specific class and it does work with phpstorm and phpstan. Meaning that I get a full list of the methods available as well as checks against bad/deprecated method calls.